src/models/dataStore.ts

```diff
@@ -5,21 +5,25 @@ import { TableFilters, TableList, TablePagination } from "../types/d2-ui-compone
 const dataStoreNamespace = "metatada-synchronization";
 
 async function getOrCreateNamespace(d2: D2): Promise<any> {
-    const existsNamespace = await d2.dataStore.has(dataStoreNamespace);
-    if (!existsNamespace) {
-        return await d2.dataStore.create(dataStoreNamespace);
-    } else {
+    try {
         return await d2.dataStore.get(dataStoreNamespace);
+    } catch (e) {
+        if (e.httpStatusCode === 404) return await d2.dataStore.create(dataStoreNamespace);
+        throw new Error(`Unhandled error with dataStore namespace ${dataStoreNamespace}`);
     }
 }
 
 async function getDataStore(d2: D2, dataStoreKey: string, defaultValue: any = []): Promise<any> {
```

Comment:
I guess that `[]` was a convenience so the callers are unchanged, but it's a bit weird to have a default value of array (typically: `undefined`).